### DISCUSSION TITLE
Inconsistent behaviour with trailing "/" in template path

### DISCUSSION THREAD

**Discussion by EAAZ02** [projectdiscovery/nuclei#1172] [Q&A] [answered]
**Describe the bug**
I have noticed that when a scan is launched with a path for the templates ending in a "/" sometimes it does not show all the "positive" templates, causing false negatives results, launching the same scan without the trailing "/" will report the correct results.

**Nuclei version**
2.5.2

**Screenshot of the error or bug**
Nuclei with trailing "/" in template path:
![imagen](https://user-images.githubusercontent.com/7225206/137481248-47bda401-6bba-49f3-9481-240a58aa2851.png)

Nuclei without the trailing "/" in template path:
![imagen](https://user-images.githubusercontent.com/7225206/137481313-dc1d5bee-dfc5-4eb1-aecd-311b0d46f0ac.png)

Another site, same weird behaviour:
![imagen](https://user-images.githubusercontent.com/7225206/137482064-cfe1a65a-5275-41c7-b289-f7aee47e7562.png)

**Comment by ehsandeep** [comment]
@EAAZ02 This appears to be more consistent behavior than inconsistent behavior; inserting a slash (`/`) in your template with less result indicates that the website you are providing as input behaves differently; you can validate this by running these templates with the `-debug` flag enabled.

In general, you should not include a slash (`/`) in the template if it's not required for the detection.

**Comment by EAAZ02** [comment]
> 
> 
> @EAAZ02 This appears to be more consistent behavior than inconsistent behavior; inserting a slash (`/`) in your template with less result indicates that the website you are providing as input behaves differently; you can validate this by running these templates with the `-debug` flag enabled.
> 
> In general, you should not include a slash (`/`) in the template if it's not required for the detection.

the "/" should not interfere in the template, it is in the system path of the template, not in the url,  it should be like using "nuclei http.... -t /some/path/to//template.yaml", which should translate into "nuclei http.... -t "/some/path/to/template.yaml", right?

If it is the intended behavior, it's ok, but seems a bit weird to me.

Thanks for replying :)

**Comment by ehsandeep** [comment, marked as answer]
@EAAZ02 I misunderstood your question, and this is definitely weird, input `/` should not affect the template results in any way, even as per the screenshot all the templates are loaded in both cases and I'm unable to replicate it.

```console
echo https://google.com | nuclei -t technologies -stats

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   2.5.3-dev

		projectdiscovery.io

[WRN] Use with caution. You are responsible for your actions.
[WRN] Developers assume no liability and are not responsible for any misuse or damage.
[INF] Using Nuclei Engine 2.5.3-dev (development)
[INF] Using Nuclei Templates 8.5.9 (latest)
[INF] Using Interactsh Server https://interactsh.com
[INF] Templates added in last update: 37
[INF] Templates loaded for scan: 199
[INF] Templates clustered: 95 (Reduced 88 HTTP Requests)
[favicon-detection:google] [http] [info] https://google.com/favicon.ico
[0:00:05] | Templates: 199 | Hosts: 1 | RPS: 36 | Matched: 1 | Errors: 0 | Requests: 183/204 (89%)
[0:00:07] | Templates: 199 | Hosts: 1 | RPS: 29 | Matched: 1 | Errors: 0 | Requests: 203/204 (99%)

nuclei-templates $echo https://google.com | nuclei -t technologies/ -stats

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   2.5.3-dev

		projectdiscovery.io

[WRN] Use with caution. You are responsible for your actions.
[WRN] Developers assume no liability and are not responsible for any misuse or damage.
[INF] Using Nuclei Engine 2.5.3-dev (development)
[INF] Using Nuclei Templates 8.5.9 (latest)
[INF] Using Interactsh Server https://interactsh.com
[INF] Templates added in last update: 37
[INF] Templates loaded for scan: 199
[INF] Templates clustered: 95 (Reduced 88 HTTP Requests)
[favicon-detection:google] [http] [info] https://google.com/favicon.ico
[0:00:05] | Templates: 199 | Hosts: 1 | RPS: 36 | Matched: 1 | Errors: 0 | Requests: 185/204 (90%)
[0:00:06] | Templates: 199 | Hosts: 1 | RPS: 32 | Matched: 1 | Errors: 0 | Requests: 203/204 (99%)
```

@EAAZ02 again can you run both scans with `debug` flag to see what's going on? also you can join https://discord.gg/projectdiscovery if you can share more information in DM.

**Reply by EAAZ02** [reply]
Sorry for the delay, it has been a busy week.

I have tried to replicate the issue with debug flag, but i couldn't. Anyway, in the 2.5.3 release seems to work as expected:

![imagen](https://user-images.githubusercontent.com/7225206/138422210-4eb4a07a-9126-48ce-8dff-a1aa1a1c475f.png)

 Thanks for your attention!

**Reply by ehsandeep** [reply]
thanks for confirming.
